File: ITServiceDesk.Core/Entities/Ticket.cs
using ITServiceDesk.Core.Enums;

namespace ITServiceDesk.Core.Entities;

public class Ticket : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    
    public TicketStatus Status { get; set; } = TicketStatus.Open;
    public Priority Priority { get; set; } = Priority.Medium;
    
    public DateTime? ResolvedAt { get; set; }
    public DateTime? ClosedAt { get; set; }
    
    // SLA Fields
    public DateTime? ResponseDueDate { get; set; }
    public DateTime? ResolutionDueDate { get; set; }
    public bool IsEscalated { get; set; } = false;
    
    // Status Report
    public string? ResolutionReport { get; set; }
    
    // Navigation Properties
    public Guid RequesterId { get; set; }
    public AppUser? Requester { get; set; }
    
    public Guid? AssigneeId { get; set; }
    public AppUser? Assignee { get; set; }
    
    public Guid? DepartmentId { get; set; }
    public Department? Department { get; set; }
    
    public ICollection<Comment> Comments { get; set; } = [];
    public ICollection<Attachment> Attachments { get; set; } = [];
    public ICollection<AuditLog> AuditLogs { get; set; } = [];
    
    public Guid? DeviceId { get; set; }
    public Device? Device { get; set; }
}

File: ITServiceDesk.Client/src/types/ticket.ts
export enum TicketStatus {
    Open = 1,
    InProgress = 2,
    WaitingForUser = 3,
    Resolved = 4,
    Closed = 5
}

export enum Priority {
    Low = 1,
    Medium = 2,
    High = 3,
    Critical = 4
}

export interface TicketResponseDto {
    id: string;
    title: string;
    description: string;
    status: TicketStatus;
    priority: Priority;
    createdAt: string;
    resolvedAt: string | null;
    responseDueDate: string | null;
    resolutionDueDate: string | null;
    isEscalated: boolean;
    requesterId: string;
    requesterName?: string;
    assigneeId: string | null;
    departmentId: string | null;
    departmentName?: string;
    deviceId: string | null;
    deviceName?: string;
    resolutionReport: string | null;
}

export interface TicketCreateDto {
    title: string;
    description: string;
    priority: Priority;
    departmentId?: string | null;
    deviceId?: string | null;
}

export interface TicketUpdateDto {
    id: string;
    title: string;
    description: string;
    status: TicketStatus;
    priority: Priority;
    assigneeId?: string | null;
    departmentId?: string | null;
    deviceId?: string | null;
    resolutionReport: string | null;
}

export interface TicketFilterDto {
    pageNumber: number;
    pageSize: number;
    status?: TicketStatus;
    priority?: Priority;
    deviceId?: string;
}
